ludwig/modules/attention_modules.py

```diff
@@ -132,20 +172,23 @@ def __init__(
         super().__init__()
         self.supports_masking = True
 
-        self.layers = []
+        self.layers = nn.ModuleList()
+
         for _ in range(num_layers):
             layer = TransformerBlock(
+                input_size=input_size,
```

Comment:
As I understand it, I believe it is correct.  This is how I ended up with the current solution.

These two sections of the TF1 code are relevant to this discussion:
*  Part of [constructor for `StackedTransfomer`](https://github.com/ludwig-ai/ludwig/blob/b15c49e7913129fa2b93d44ec9ab19d69aeffc58/ludwig/encoders/sequence_encoders.py#L2020-L2051). Lines 2035 to 2042 sets up a layer called `self.project_to_hidden_size` and sets the flag `self.should_project = True`
* Beginning part of the [`call()` method for `StackedTransformer`](https://github.com/ludwig-ai/ludwig/blob/b15c49e7913129fa2b93d44ec9ab19d69aeffc58/ludwig/encoders/sequence_encoders.py#L2087-L2109).   As best I can tell `self.should_project` will be True when `call()` method is invoked.  This means the layer `self.project_to_hidden_size()` will be invoked to create a tensor whose last dimension is `hidden_size`.

So when `self.transformer_stack()` is called, the input tensor (aka `hidden`) has the same last dimension as the hidden layers in Transformer Stack.  In other words, "the input size for the first one" is `hidden_size`, which is the same "hidden size for the following blocks."

Did this answer the question?

Reply:
i see your point, and your reasoning is correct wrt the `StackedTransfomer` behavior.
That said, the `TransformerStack` should likely have a parameter for inputs_size and one for hidden_size. In most cases they will be the same, in particular when called by the `StackedTransfomer`, but there could be another Module that doesn't perform the projection beforehand, so the first block in the stack of transformers will have hidden size as output and the following one will keep on using hidden size as input size. Does this make sense?

Reply:
Got it....I understand for generality the first layer to be the `input_size` and the subsequent layers are `hidden_size` and not assume from the current implementation that this will always be the case.

Reply:
@w4nderlust   Just pushed commit that addresses your comment.